aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -243,11 +243,28 @@ Tensor& ceil_out(Tensor& result, const Tensor& self) {
   // Note: this is consistent with NumPy
   TORCH_CHECK(!self.is_complex(),
     "ceil is not supported for complex inputs");
+  if (c10::isIntegralType(self.scalar_type(), /*include_bool=*/false)) {
```

Comment:
While NumPy promotes integers to float, Python does not. cc @rgommers. Given that discrepancy I'm not sure it's worth mimicking the NumPy behavior.

From an implementation perspective, we have helpers like unary_op_impl_float_out that implement this behavior. See, for example, the implementation of the exp() operator below.

Reply:
I'd vote against mimicking numpy behavior here. If we decide to cast to float, it probably should be pytorch's default fp type, which is fp32, not fp64. But casting to fp32 is imprecise even for relatively small integer numbers.

Reply:
Hmm, but does it make sense that `torch.ceil` and `torch.floor` should be essentially a no-ops for all integral types(including bools?)

Reply:
Let's give @rgommers a chance to weigh-in. We should either return self for integral types (including bools) or continue to not support those dtypes. If we return self (consistent with Python) we should file a bug with NumPy, too, for the same behavior.

Reply:
Python stdlib behaviour I usually don't pay much attention to, only built-in operators. In this case `math` always casts to `int`, which is a decent choice too I think. It all doesn't matter too much. 
```
>>> type(math.floor(2))                                                     
<class 'int'>
>>> type(math.floor(2.5))                                                   
<class 'int'>
>>> np.floor(np.arange(3, dtype=np.int64)).dtype                            
dtype('float64')
>>> np.floor(np.arange(3., dtype=np.float64)).dtype                         
dtype('float64')
```

As long as the design is consistent, I think exact numpy compat isn't that important for this kind of thing. I agree with @ngimel's comment here (casting int to float should yield the default `float32`), so I'd stay with the "preserve same dtype" design rule for `ceil` and `floor`.